modules/ximgproc/src/radon_transform.cpp

```diff
@@ -0,0 +1,421 @@
+  {
+    setMinMaxAngle(minAngle, maxAngle, 180, 180);
+  }
+}
+
+template <typename T, int D>
+static void rT_T( Mat srcMat,
+                  Mat & dstMat,
+                  int minAngle,
+                  int maxAngle
+                ) {
+  int rows = srcMat.rows, cols = srcMat.cols;
+  int m = cols/2, n = rows/2;
+
+  for(int i = minAngle; i <= maxAngle; ++i) {
+    trans<T, D>(srcMat, dstMat, m, n, dstMat.rows - 1, i, minAngle);
+  }
+}
+
+static void rT( Mat src,
+                Mat & dst,
+                int minAngle,
+                int maxAngle
+              ) {
+  int const depth = src.depth();
+  switch (depth) {
+    case CV_8U:
+      rT_T<uchar, CV_8UC1>(src, dst, minAngle, maxAngle);
+      break;
+    case CV_8S:
+      rT_T<schar, CV_8SC1>(src, dst, minAngle, maxAngle);
+      break;
+    case CV_16U:
+      rT_T<ushort, CV_16UC1>(src, dst, minAngle, maxAngle);
+      break;
+    case CV_16S:
+      rT_T<short, CV_16SC1>(src, dst, minAngle, maxAngle);
+      break;
+    case CV_32S:
+      rT_T<int, CV_32SC1>(src, dst, minAngle, maxAngle);
+      break;
+    case CV_32F:
+      rT_T<float, CV_32FC1>(src, dst, minAngle, maxAngle);
+      break;
+    case CV_64F:
+      rT_T<double, CV_64FC1>(src, dst, minAngle, maxAngle);
+      break;
+    default:
+      CV_Error_(CV_StsNotImplemented, ("Unknown depth %d", depth));
+      break;
+  }
+}
+
+/**
+ * Creates the destination image with appropriate size, channels and depth
+ */
+static void createDstRadonMat( OutputArray dst,
+                               InputArray src,
+                               int depth,
+                               int minAngle,
+                               int maxAngle
+                             ) {
+  int const rows = src.size().height;
+  int const cols = src.size().width;
+  int const channels = src.channels();
+
+  if(depth == -1) {
+    depth = src.depth();
+  }
+
+  int rhoMax = static_cast<int>(std::ceil(std::sqrt(rows*rows + cols*cols)));
+  int lenTheta = maxAngle - minAngle + 1;
+
+  dst.create(Size(lenTheta, rhoMax + 1), CV_MAKETYPE(depth, channels));
+}
+
+/**
+ * [radonTransform Computes the Radon transform of the given image]
+ * @param src        [Input source image]
+ * @param dst        [Output image]
+ * @param angleRange [One of the enums, RadonAngleRange, defaults to 63 ( 1 to 180 degrees)]
+ * @param operation  [One of the enums, RadonOp, defaults to RT_SUM ( SUM OPERATION )]
+ * @param opDst      [Output image of the operation perfomed, defaults to noArray()]
+ */
+void radonTransform( InputArray src,
+                     OutputArray dst,
+                     int angleRange,
+                     int operation,
+                     OutputArray opDst
+                   ) {
+  /**
+   * Asserting proper width, height and channels.
+   * Input should be a one channel image.
+   */
+  CV_Assert(src.size().width > 0 && src.size().height > 0);
+  CV_Assert(src.channels() == 1);
+
+  Mat srcMat = src.getMat().clone();
```

Comment:
`trans(..)` function doesn't modify source matrix so it isn't necessary to clone `src` here. However since `trans(..)` could be called and therefore iterate over the same points several times for different angles probably it make sense to convert `src` to CV_32FC1 here instead of casting its content to float inside the `trans(..)`